include/circt/Dialect/FIRRTL/FIRParser.h

```diff
@@ -78,41 +78,41 @@ void registerFromFIRFileTranslation();
 
 /// The FIRRTL specification version.
 struct FIRVersion {
-  uint32_t major, minor, patch;
-
-  /// Three way compare of one FIRRTL version with another FIRRTL version.
-  /// Return 1 if the first version is greater than the second version, -1 if
-  /// the first version is less than the second version, and 0 if the versions
-  /// are equal.
-  static int compare(const FIRVersion &a, const FIRVersion &b) {
-    if (a.major > b.major)
-      return 1;
-    if (a.major < b.major)
-      return -1;
-    if (a.minor > b.minor)
-      return 1;
-    if (a.minor < b.minor)
-      return -1;
-    if (a.patch > b.patch)
-      return 1;
-    if (a.patch < b.patch)
-      return -1;
-    return 0;
+  constexpr FIRVersion(uint16_t major, uint16_t minor, uint16_t patch)
+      : major(major), minor(minor), patch(patch) {}
+
+  explicit constexpr operator uint64_t() const {
+    return uint64_t(major) << 32 | uint64_t(minor) << 16 | uint64_t(patch);
+  }
+
+  constexpr bool operator<(FIRVersion rhs) const {
+    return uint64_t(*this) < uint64_t(rhs);
+  }
+
+  constexpr bool operator>(FIRVersion rhs) const {
+    return uint64_t(*this) > uint64_t(rhs);
   }
 
-  static FIRVersion minimumFIRVersion() { return {0, 2, 0}; }
+  constexpr bool operator<=(FIRVersion rhs) const {
+    return uint64_t(*this) <= uint64_t(rhs);
+  }
 
-  static FIRVersion defaultFIRVersion() { return {1, 0, 0}; }
+  constexpr bool operator>=(FIRVersion rhs) const {
+    return uint64_t(*this) >= uint64_t(rhs);
+  }
 
-  static FIRVersion latestFIRVersion() { return {3, 1, 0}; }
+  uint16_t major;
+  uint16_t minor;
+  uint16_t patch;
+};
 
-}; // namespace firrtl
+constexpr FIRVersion minimumVersion(0, 2, 0);
+constexpr FIRVersion latestVersion(3, 1, 0);
+constexpr FIRVersion defaultVersion(1, 0, 0);
```

Comment:
These names are a bit generic. However, they are under the `firrtl` namespace. That seems fine.

Reply:
I agree, renamed them back to minimumFIRVersion etc, but left them out of the struct. I had to move them to after the class definition so that they could be constexpr.